src/map/script.cpp

```diff
@@ -11061,6 +11061,118 @@ BUILDIN_FUNC(getareausers)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/*==========================================
+ * getunits(<type>{,<array_variable>})
+ * getmapunits(<type>,<"map name">{,<array_variable>})
+ * getareaunits(<type>,<"map name">,<x1>,<y1>,<x2>,<y2>{,<array_variable>})
+ *
+ * type :
+ *	UNITTYPE_PC
+ *	UNITTYPE_NPC
+ *	UNITTYPE_MOB
+ *	UNITTYPE_PET
+ *	UNITTYPE_HOM
+ *	UNITTYPE_MER
+ *	UNITTYPE_ELEM
+ *
+ * if array_variable is set
+ *	int array_variable will return GID List
+ *	string array_variable will return names List
+ *
+ *	return the count of the type given
+ *------------------------------------------*/
+BUILDIN_FUNC(getunits)
+{
+	struct block_list *bl = NULL;
+	struct s_mapiterator *iter = NULL;
+	struct map_session_data *sd = NULL;
+	struct script_data* data = NULL;
+	char *command = (char *)script_getfuncname(st);
+	const char *str;
+	const char *name;
+	int type = script_getnum(st, 2);
+	int size = 0;
+	int32 idx, id;
+	int16 m = 0, x0 = 0, y0 = 0, x1 = 0, y1 = 0;
+
+	switch (type)
+	{
+		case UNITTYPE_PC: iter = mapit_geteachpc(); break;
+		case UNITTYPE_NPC: iter = mapit_geteachnpc(); break;
+		case UNITTYPE_PET: iter = mapit_geteachpet(); break;
+		case UNITTYPE_MOB: iter = mapit_geteachmob(); break;
+		case UNITTYPE_HOM: iter = mapit_geteachhom(); break;
+		case UNITTYPE_MER: iter = mapit_geteachmer(); break;
+		case UNITTYPE_ELEM: iter = mapit_geteachelem(); break;
```

Comment:
Add support for real `BL_` types. Don't forget you will have to export `BL_NPC` in `script_constants.hpp`.